app/Models/Group.php

```diff
@@ -55,25 +57,23 @@ function members()
 
     #endregion
 
-    #region Attributes
+    #region Scopes
 
-    function getDefaultStateAttribute()
+    public function scopeHasMember(Builder $query, $member)
     {
-        if ($this->state) {
-            return $this->state;
-        }
-
-        for ($parent = $this->parent; $parent; $parent = $parent->parent)
-        {
-            if ($parent->state) {
-                return $parent->state;
-            }
-        }
+        return $query
+            ->whereHas('members', function(Builder $query) use($member){
+                $query->where('user_id', $member->id);
+            });
     }
-
-    function getParticipantsAttribute()
+    public function scopeHasBookmarked(Builder $query, $bookmarkable_type, $bookmarkable_id)
     {
-        return $this->members->pluck('user')->prepend($this->owner)->unique();
+        return $query
+            ->whereHas('workspaces.topics.bookmarks', function(Builder $query) use($bookmarkable_type, $bookmarkable_id){
+                $query->whereHasMorph('bookmarkable', $bookmarkable_type, function(Builder $query) use($bookmarkable_id){
```

Comment:
I honestly don't even want to try to figure out what this is doing to even suggest a better option. It's that gross. My brain can't acknowledge that it really exists.

Reply:
I think it's fixed now... somewhat.